api/client/build.go

```diff
@@ -210,6 +211,18 @@ func (cli *DockerCli) CmdBuild(args ...string) error {
 		}
 	}
 
+	var shmSize int64 = 67108864 // initial SHM size is 64MB
+	if *flShmSize != "" {
+		parsedShmSize, err := units.RAMInBytes(*flShmSize)
+		if err != nil {
+			return err
+		}
+		if parsedShmSize <= 0 {
+			return fmt.Errorf("--shm-size: 0 or less cannot be specified for SHM size")
```

Comment:
`--shm-size: SHM size must be greater than 0 . You specified: %parsedShmSize`


Better to say in the positive if that works...my syntax for referring to their string is no doubt bogus but you get the idea.